java_src/foam/dao/MapDAO.java

```diff
@@ -51,9 +51,9 @@ public FObject remove(FObject obj) {
 
   public FObject find(Object o) {
     if( getOf().getObjClass().isInstance(o) ) {
-      return getData().get(getPrimaryKey().get(o));
+      return getData().get(getPrimaryKey().get(o)).fclone();
```

Comment:
Rewrite as:

return maybeClone(getOf().getObjClass().isInstance(o) ?
    getData().get(getPrimaryKey().get(o)):
    getData.get(o));

where maybeClone() is a method that you add which clones if not null.
isInstance() could also be added to ClassInfo so you could just do getOf().isInstance()